### ISSUE TITLE
cli tool results in ERROR: Invalid range specified: (-1,-1);

### ISSUE THREAD

**Issue by snowe2010** [arturbosch/detekt#422]
No matter how I run the cli tool (after building locally) I get this error. 
I've tried with and without all of these parameters. 

Attempting to supply the `--debug` option results in the config file being printed and that's it. 

```
❯ java -jar /Users/tylerthrailkill/Documents/dev/code/detekt/detekt-cli/build/libs/detekt-cli-1.0.0.RC4-3-all.jar -i . -o . -on kotlinreports -c default-detekt-config.yml  
Analyzing 514 kotlin files: ...............................................................................................................................ERROR: Invalid range specified: (-1,-1);
java.lang.Throwable
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.startLineAndColumn(Location.kt:38)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.from(Location.kt:25)
	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:24)
	at io.gitlab.arturbosch.detekt.rules.style.NewLineAtEndOfFile.visitKtFile(NewLineAtEndOfFile.kt:22)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:513)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:246)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:226)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visit(BaseRule.kt:22)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visitFile(BaseRule.kt:16)
	at io.gitlab.arturbosch.detekt.api.RuleSet.accept(RuleSet.kt:23)
	at io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:54)
	at io.gitlab.arturbosch.detekt.core.Detektor.access$analyze(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:32)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.TasksKt$runAsync$1.invoke(Tasks.kt:20)
	at io.gitlab.arturbosch.detekt.core.TasksKt$task$1.get(Tasks.kt:24)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
............................................................Exception in thread "main" java.util.concurrent.CompletionException: java.lang.AssertionError: Invalid range specified: (-1,-1);
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.AssertionError: Invalid range specified: (-1,-1);
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:71)
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.startLineAndColumn(Location.kt:38)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.from(Location.kt:25)
	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:24)
	at io.gitlab.arturbosch.detekt.rules.style.NewLineAtEndOfFile.visitKtFile(NewLineAtEndOfFile.kt:22)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:513)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:246)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:226)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visit(BaseRule.kt:22)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visitFile(BaseRule.kt:16)
	at io.gitlab.arturbosch.detekt.api.RuleSet.accept(RuleSet.kt:23)
	at io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:54)
	at io.gitlab.arturbosch.detekt.core.Detektor.access$analyze(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:32)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.TasksKt$runAsync$1.invoke(Tasks.kt:20)
	at io.gitlab.arturbosch.detekt.core.TasksKt$task$1.get(Tasks.kt:24)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	... 3 more
Caused by: java.lang.Throwable
	... 27 more
....................................................................................................................................................................................................................................................................ERROR: Invalid range specified: (-1,-1);
java.lang.Throwable
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.startLineAndColumn(Location.kt:38)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.from(Location.kt:25)
	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:24)
	at io.gitlab.arturbosch.detekt.rules.style.NewLineAtEndOfFile.visitKtFile(NewLineAtEndOfFile.kt:22)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:513)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:246)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:226)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visit(BaseRule.kt:22)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visitFile(BaseRule.kt:16)
	at io.gitlab.arturbosch.detekt.api.RuleSet.accept(RuleSet.kt:23)
	at io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:54)
	at io.gitlab.arturbosch.detekt.core.Detektor.access$analyze(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:32)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.TasksKt$runAsync$1.invoke(Tasks.kt:20)
	at io.gitlab.arturbosch.detekt.core.TasksKt$task$1.get(Tasks.kt:24)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
.........ERROR: Invalid range specified: (-1,-1);
java.lang.Throwable
	at org.jetbrains.kotlin.com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at org.jetbrains.kotlin.com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.startLineAndColumn(Location.kt:38)
	at io.gitlab.arturbosch.detekt.api.Location$Companion.from(Location.kt:25)
	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:24)
	at io.gitlab.arturbosch.detekt.rules.style.NewLineAtEndOfFile.visitKtFile(NewLineAtEndOfFile.kt:22)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:513)
	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:246)
	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.java:226)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visit(BaseRule.kt:22)
	at io.gitlab.arturbosch.detekt.api.BaseRule.visitFile(BaseRule.kt:16)
	at io.gitlab.arturbosch.detekt.api.RuleSet.accept(RuleSet.kt:23)
	at io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:54)
	at io.gitlab.arturbosch.detekt.core.Detektor.access$analyze(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:32)
	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:16)
	at io.gitlab.arturbosch.detekt.core.TasksKt$runAsync$1.invoke(Tasks.kt:20)
	at io.gitlab.arturbosch.detekt.core.TasksKt$task$1.get(Tasks.kt:24)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
..........................................................
```

**Comment by snowe2010** [comment]
Ok, so making

```kotlin
@Suppress("TooGenericExceptionCaught")
fun startLineAndColumn(element: PsiElement, offset: Int = 0): DiagnosticUtils.LineAndColumn {
    return try {
        val range = element.textRange
        DiagnosticUtils.getLineAndColumnInPsiFile(element.containingFile,
                TextRange(range.startOffset + offset, range.endOffset + offset))
    } catch (e: IndexOutOfBoundsException) {
        // #18 - somehow the TextRange is out of bound on '}' leaf nodes, returning fail safe -1
        DiagnosticUtils.LineAndColumn(-1, -1, null)
    }
}
```

into 

```kotlin
@Suppress("TooGenericExceptionCaught")
fun startLineAndColumn(element: PsiElement, offset: Int = 0): DiagnosticUtils.LineAndColumn {
    return try {
        val range = element.textRange
        DiagnosticUtils.getLineAndColumnInPsiFile(element.containingFile,
                TextRange(range.startOffset + offset, range.endOffset + offset))
    } catch (e: IndexOutOfBoundsException) {
        // #18 - somehow the TextRange is out of bound on '}' leaf nodes, returning fail safe -1
        DiagnosticUtils.LineAndColumn(-1, -1, null)
    } catch (e: AssertionError) {
        DiagnosticUtils.LineAndColumn(-1, -1, null)
    }
}
```

fixes the issue. 

Not sure if that is the actual solution or not, as I don't know how to tell which element it's actually breaking on.

**Comment by arturbosch** [comment]
Interesting. Could you find out in which kotlin file this happens? Analyzing package after package and inside the right package just the files? It is the `NewLineAtEndOfFile` rule which produces wrong source locations. You can turn this rule off in the config and use detekt without this rule for now. Thanks for reporting!

**Comment by snowe2010** [comment]
How can I print which file it's happening in? We have more than 500+ kotlin files.

**Comment by arturbosch** [comment]
I assume you have modules and packages? Try to set the detekt extension's input parameter to some specific module/package and see if it crashes. If it does go further and try to look at the sub-packages, later each file. In the end it must crash for a specific file. It would be nice to have the code for this file to fix the issue and not just write a workaround for the assertion error case :)

**Comment by snowe2010** [comment]
Closed and clarified in #423
